typed-racket-lib/typed-racket/static-contracts/optimize.rkt

```diff
@@ -60,6 +60,14 @@
       [(? (λ (l) (member any/sc l))) any/sc]
       [(? (λ (l) (member none/sc l)))
        (apply or/sc (remove* (list none/sc) scs))]
+      [(list pre-scs ... (or/sc: mid-scs ...) post-scs ...)
```

Comment:
Will this flatten multiple `or/sc` that appear inside the outer or? Also, doing this repeatedly and via complex `...` patterns seems likely to be less efficient than just writing a recursive function directly.